app/domain/authentication/authn_oidc/v2/commands/list_providers.rb

```diff
@@ -10,31 +13,24 @@ module Commands
         ListProviders ||= CommandClass.new(
           dependencies: {
             json_lib: JSON,
-            logger: Rails.logger
+            logger: Rails.logger,
+            authenticatorRepository: nil,
+            provider: Authentication::AuthnOidc::V2::Views::ProviderContext.new()
           },
           inputs: %i[message]
         ) do
           def call
             params = @json_lib.parse(@message)
             @logger.debug("#{self.class}##{__method__} - #{params}")
             (account = params.delete("account")) || raise("'account' is required")
-
-            authenticators = DB::Repository::AuthenticatorRepository.new(
-                data_object: Authentication::AuthnOidc::V2::DataObjects::Authenticator,
-                logger: @logger
-              ).find_all(
-                account: account,
-                type: 'authn-oidc'
-              )
+            authenticators = @authenticatorRepository.find_all(account: account, type: 'authn-oidc')
             @logger.debug("#{self.class}##{__method__} - #{authenticators.inspect}")
-            Authentication::AuthnOidc::V2::Views::ProviderContext.new(
-              logger: @logger
```

Comment:
With moving the ProviderContext.new up to line 18, does this lose the inclusion of the logger?